go/libraries/doltcore/sqle/dsess/session_state_adapter.go

```diff
@@ -61,29 +61,30 @@ func (s SessionStateAdapter) GetRoots(ctx context.Context) (doltdb.Roots, error)
 	return state.GetRoots(), nil
 }
 
-func (s SessionStateAdapter) CWBHeadRef() ref.DoltRef {
+func (s SessionStateAdapter) CWBHeadRef() (ref.DoltRef, error) {
 	workingSet, err := s.session.WorkingSet(sql.NewContext(context.Background()), s.dbName)
 	if err != nil {
-		// TODO: fix this interface
-		panic(err)
+		return nil, err
```

Comment:
This is where the error should originate, not session.WorkingSet

Reply:
In principle, I agree with your comments. It should be up to the client to decide whether the lack of a working set is valid, and up the api to return results in a form that require clients to consider this.

The trouble is that a lot of places call session.Workingset without calling CWBHeadRef, and don't do any checking on the result, and adding checks to all of them would complicate all those commands while doing nothing to prevent this same mistake from happening if we add another command that calls session.Workingset.

The long term solution is to change the interface to one where instead of clients requesting the working set, they request what they want the working set for, and the and session returns the best result. But until we have that, the priority is to prevent panics and make it harder to introduce panics in the future. This does that.

Reply:
I'm fine with this for now, please add a TODO to this effect in the method doc for Database.GetWorkingSet()

Reply:
Done